tests/Metadata/Operation/Factory/OperationMetadataFactoryTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Tests\Metadata\Operation\Factory;
+
+use ApiPlatform\Metadata\ApiResource;
+use ApiPlatform\Metadata\Get;
+use ApiPlatform\Metadata\Operation\Factory\OperationMetadataFactory;
+use ApiPlatform\Metadata\Resource\Factory\ResourceMetadataCollectionFactoryInterface;
+use ApiPlatform\Metadata\Resource\Factory\ResourceNameCollectionFactoryInterface;
+use ApiPlatform\Metadata\Resource\ResourceMetadataCollection;
+use ApiPlatform\Metadata\Resource\ResourceNameCollection;
+use PHPUnit\Framework\TestCase;
+use Prophecy\PhpUnit\ProphecyTrait;
+
+/**
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+class OperationMetadataFactoryTest extends TestCase
+{
+    use ProphecyTrait;
+
+    public function testCreate(): void
+    {
+        $resourceNameCollectionFactory = $this->prophesize(ResourceNameCollectionFactoryInterface::class);
+        $resourceMetadataCollectionFactory = $this->prophesize(ResourceMetadataCollectionFactoryInterface::class);
+        $resourceNameCollectionFactory->create()->willReturn(new ResourceNameCollection(['one']));
+        $resourceMetadataCollectionFactory->create('one')->willReturn(new ResourceMetadataCollection('one', [
+            new ApiResource(operations: [new Get('/one', name: 'one')]),
+        ]));
+
+        $operationMetadata = new OperationMetadataFactory($resourceNameCollectionFactory->reveal(), $resourceMetadataCollectionFactory->reveal());
+        $this->assertEquals(new Get('/one', name: 'one'), $operationMetadata->create('one'));
```

Comment:
Shouldn't we use `assertSame` instead of `assertEquals`, and store the `new Get` operation in a local variable, to ensure it's not a new one or a copy?

Reply:
they should reference the same object, I'll however reduce the amount of code using a variable.